app/controllers/course/announcements_controller.rb

```diff
@@ -51,4 +52,18 @@ def destroy #:nodoc:
   def announcement_params #:nodoc:
     params.require(:announcement).permit(:title, :content, :sticky, :valid_from, :valid_to)
   end
+
+  # Ensure that the component is enabled and load its settings.
+  #
+  # @raise [ActionController::RoutingError] When the component is disabled.
+  def check_component_and_load_settings
+    announcement_component = current_component_host[:course_announcements_component]
+    raise ActionController::RoutingError.
```

Comment:
Provide an exception class and message as arguments to `raise`.<br>Use `fail` instead of `raise` to signal exceptions.